service/history/workflow/activity.go

```diff
@@ -76,3 +82,92 @@ func updateActivityInfoForRetries(
 
 	return ai
 }
+
+func ActivityState(ai *persistence.ActivityInfo) enumspb.PendingActivityState {
+	if ai.CancelRequested {
+		return enumspb.PENDING_ACTIVITY_STATE_CANCEL_REQUESTED
+	}
+	if ai.StartedEventId != common.EmptyEventID {
+		return enumspb.PENDING_ACTIVITY_STATE_STARTED
+	}
+	return enumspb.PENDING_ACTIVITY_STATE_SCHEDULED
+}
+
+func GetPendingActivityInfo(
```

Comment:
Can we reuse this function in the describe implementation? I would rather not duplicate code unless strictly necessary.

Reply:
yes, that was the whole idea.
I will need it in "describe activity" API.
 I even did that, but looks like didn't save or something.